### PR TITLE
KAFKA-15852: Move LinuxIoMetricsCollector to server module

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server/src/main/java/org/apache/kafka/server/metrics/LinuxIoMetricsCollector.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+     * cancelled_write_bytes: 0
+     */
+    private boolean updateValues(long now) {
+        synchronized (this) {
+            try {
+                cachedReadBytes = -1L;
+                cachedWriteBytes = -1L;
+                List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+                lines.forEach(line -> {
+                    synchronized (this) {
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/server/metrics/LinuxIoMetricsCollector.java]: Why we need two `synchronized (this)`?

File: core/src/main/scala/kafka/metrics/LinuxIoMetricsCollector.scala [hunk 2]
```diff
@@ -1,101 +0,0 @@
...
-
-import java.nio.file.{Files, Path, Paths}
-import org.apache.kafka.common.utils.Time
-import org.slf4j.Logger
-
-import java.nio.charset.StandardCharsets
-import scala.jdk.CollectionConverters._
-
-/**
- * Retrieves Linux /proc/self/io metrics.
```
Comment (chia7712) [inline core/src/main/scala/kafka/metrics/LinuxIoMetricsCollector.scala:28]: Maybe we can keep this comment?

**Review by chia7712 — COMMENTED** [review]
@mimaison thanks for this patch.

File: server/src/main/java/org/apache/kafka/server/metrics/LinuxIoMetricsCollector.java (hunk 1 above)
Comment (mimaison) [reply server/src/main/java/org/apache/kafka/server/metrics/LinuxIoMetricsCollector.java]: This was inside a lambda. I refactored this bit to simply use a loop so it's not needed anymore.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by mimaison** [comment]
Test failures are not related, merging to trunk.

**Pull request merged**